contracts/Smt.sol

```diff
@@ -14,7 +14,10 @@ contract Smt is OwnableUpgradeable {
      * @dev Throws if called by any account other than the state contract.
      */
     modifier onlyWriter() {
-        require(_writer == _msgSender(), "caller has no permissions");
+        require(
+            owner() == _msgSender() || _writer == _msgSender(),
```

Comment:
Granting the owner the power to write to the tree is something that we should  consider very carefully. An owner can now change any state on his own (anytime, not only during the migration) Looks like too much power to him, as he can "hack" any identity proof in that way by substituting its state.
Consider modifying permissions so to grant the owner as minimal as possible